chapters/en/slides/module3/module3_01.Rmd

```diff
@@ -0,0 +1,274 @@
+
+Before diving into these comparisons,
+we want to understand the data for each of the columns in our dataset.
+
+Let's start by looking at the ratings the movies received
+and explore the best ways to visualize this data.
+
+---
+
+## Visualizing a single column with a point plot along one axis
+
+```{python}
+import altair as alt
+
+
+alt.Chart(movies[:5]).mark_point().encode(
+    x='runtime')
+```
+
+Notes:
+One of the simplest visualization we could perform
+with a single dataframe column
+is to plot each individual observations
+as a point.
+
+This is similar to a scatter plot,
+but only using one of the plot axes.
+
+It works well when there are just a few plots to visualize,
+which is why we subset the data in this slide.
+
+---
+
+## Single axis point plots can become saturated when there is a lot of data to plot
+
+```{python}
+alt.Chart(movies[:50]).mark_point().encode(
+    x='runtime')
+```
+
+Notes:
+When there are more observations,
+it becomes really hard to see which ones are where
+since the circular marks are overlapping with each other.
+
+When it becomes hard to tell individual observations apart,
+we say that a plot become saturated.
+We will talk more about this concept later in the module.
+
+---
+
+## Thinner marks avoids saturation
+
+```{python}
+alt.Chart(movies[:50]).mark_tick().encode(
+    x='runtime')
+```
+
+Notes:
+It is often easier to separate individual observations
+when they are represented as thin tick marks instead of the thick points
+as in the previous slide.
+
+The disadvantage of this plot type
+is that it can be a bit harder to see individual observations.
+However,
+when we are visualization many marks,
+we are often more interested in how the data is shaped (or distributed) in general,
+rather than looking at individual points.
+
+This type of plot with vertical tick marks along one axis
+is so common that it is sometimes referred to by a special name: rugplot.
+
+The visualization grammar of Altair let's us think about it more intuitively
+by composing it with a mark and a single encoding,
+instead of remembering a specific name.
+
+---
+
+## Thinner marks also saturate for large data sets
+
+```{python}
+alt.Chart(movies).mark_tick().encode(
+    x='runtime')
+```
+
+Notes:
+However,
+when we increase the number of observations even further (to about 500)
+it becomes difficult to tell exactly how the data distribution looks,
+even with these thin marks.
+
+---
+
+## Using histograms to visualize distributions avoids saturation issues
+
+```{python}
+alt.Chart(movies).mark_bar().encode(
+    alt.X('runtime', bin=alt.Bin(maxbins=30)),
+    y='count()')
+```
+
+Notes:
+Instead of visualizing each individual observation,
+we can use the histograms we learnt about in last module,
+where we divide one axis into groups (or "bins")
+and represent the count of observations in each group with bar.
+
+In this plot,
+we can directly see that most movies are around 100 min in length.
```

Comment:
Add comments on range of the data - what is the minimum and maximum movie length. This is something visualizing distributions lets us know. Also, what about the shape? Are the values uniformly distributed? Or do most values fall in one peak in the middle? Also say why you can say that most movies are around 100 min in length (because there is a single peak, meaning most values in the data set fall there, and that peak is centered around 100).

Then take a breath and start discussing multiple distributions.